tutorials/raw/plot_annotating_raw.py

```diff
@@ -0,0 +1,251 @@
+#     annotations). A good safeguard is to run
+#     ``interactive_annot = raw.annotations`` after you finish an interactive
+#     annotation session, so that the annotations are stored in a separate
+#     variable outside the :class:`~mne.io.Raw` object.
+#
+#
+# How annotations affect preprocessing and analysis
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# You may have noticed that the description for new labels in the annotation
+# controls window defaults to ``BAD_``. The reason for this is that annotation
+# is often used to mark bad temporal spans of data (such as movement artifacts
+# or environmental interference that cannot be removed in other ways such as
+# projection or filtering). Several MNE-Python operations
+# are "annotation aware" and will avoid using data that is annotated with a
+# description that begins with "bad" or "BAD"; such operations typically have a
+# boolean ``reject_by_annotation`` parameter. Examples of such operations are
+# independent components analysis (:class:`mne.preprocessing.ICA`), functions
+# for finding heartbeat and blink artifacts
+# (:func:`~mne.preprocessing.find_ecg_events`,
+# :func:`~mne.preprocessing.find_eog_events`), and creation of epoched data
+# from continuous data (:class:`mne.Epochs`).
+#
+# .. TODO: change "projection" in prev. para to this when tut is ready:
+#    :ref:`projection <ssp-tutorial>`
+#
+#
+# Operations on Annotations objects
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# :class:`~mne.Annotations` objects can be combined by simply adding them with
+# the ``+`` operator, as long as they share the same ``orig_time``:
+
+new_annot = mne.Annotations(onset=3.75, duration=0.75, description='AAA')
+raw.set_annotations(my_annot + new_annot)
+raw.plot(start=2, duration=6)
+
+###############################################################################
+# Notice that it is possible to create overlapping annotations, even when they
+# share the same description. This is *not* possible when annotating
+# interactively; click-and-dragging to create a new annotation that overlaps
+# with an existing annotation with the same description will cause the old and
+# new annotations to be merged.
+#
+# Individual annotations can be accessed by indexing an
+# :class:`~mne.Annotations` object, and subsets of the annotations can be
+# achieved by either slicing or indexing with a list, tuple, or array of
+# indices:
+
+print(raw.annotations[0])       # just the first annotation
+print(raw.annotations[:2])      # the first two annotations
+print(raw.annotations[(3, 2)])  # the fourth and third annotations
+
+###############################################################################
+# You can also iterate over the annotations within an :class:`~mne.Annotations`
+# object:
+
+for ann in raw.annotations:
+    descr = ann['description']
+    start = ann['onset']
+    end = ann['onset'] + ann['duration']
+    print(f"'{descr}' goes from {start} to {end}")
```

Comment:
I like the f-string syntax a lot, but it's not compatible with python 3.5 ... and in the installation guide, we currently say:

> MNE-Python 0.18 only supports Python version 3.5 or higher.

this is "only" the documentation though, so using f-strings is probably ok? What is our general policy for string formatting in MNE for the future? Fstrings? .format syntax? % syntax? concatenation? ... whatever we like best? ...

Reply:
I would go with format

Reply:
yes no f strings in mne-python please. thanks